src/js/popup.js

```diff
@@ -469,25 +469,34 @@ $(document).ready(function () {
                     Materialize.toast('Error reading file. Please try again', 2000, 'toastNotification');
                 };
                 reader.onload = function () {
-                    var data = $.parseJSON(reader.result);
-                    chrome.runtime.sendMessage({
-                        JDTIfunction: 'upload',
-                        fileContents: data
-                    }, function (response) {
-                        if (response.status === 'success') {
-                            location.reload();
-                            Materialize.toast('Templates successfully loaded from file', 2000, 'toastNotification');
-                            dmTemplateUpdate('upload');
-                        } else {
-                            if (response.message) {
-                                dmError('upload', response.message);
-                                Materialize.toast(response.message, 2000, 'toastNotification');
+                    let validJSON = true;
```

Comment:
We're already parsing the JSON in the background page. This try/catch isn't necessary.

Reply:
:+1: